examples/mongoose.test.js

```diff
@@ -2045,4 +2045,25 @@ describe('Mongoose', function() {
     assert.equal(profilingLevel, 'off');
     // acquit:ignore:end
   });
+
+  it('validates-email', async function() {
+    const User = mongoose.model('User', mongoose.Schema({
+      email:{ 
+      type: String,
+      required: true,
+      match: /.+\@.+\..+/,
+      unique: true
+      }
+    }));
+    await User.create([
+        { email: 'gmail@google.com' },
+        { email: 'bill@microsoft.com' },
+        { email: 'test@gmail.com' }
+    ]);
+    await User.find(); // Prints the information associated with the three emails above.
+    await User.init();
+    await User.create({ email: 'gmail@google.com' }, function(err) {
```

Comment:
Don't mix promises and callbacks. I'd recommend just wrapping this `await` in a `try/catch`